src/main/java/net/masterthought/cucumber/json/deserializers/EmbeddingDeserializer.java

```diff
@@ -14,32 +14,54 @@
 import net.masterthought.cucumber.ValidationException;
 import net.masterthought.cucumber.json.Embedding;
 
+import static java.nio.charset.StandardCharsets.UTF_8;
+
 /**
  * Deserializes embedding and stores it in attachment directory.
- * 
+ *
  * @author Damian Szczepanik (damianszczepanik@github)
  */
 public class EmbeddingDeserializer extends CucumberJsonDeserializer<Embedding> {
 
+    /*
+    Base64 Regex matches for the following:
+    ^([A-Za-z0-9+/]{4})* matches group of four characters from [A-Z, a-z, 0-9, and + /]
+    If a base64 encoded String is not a multiple of four it is padded with '='.
+    (?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$ checks if the String ends with a group of four of the legal characters
+    or it ends with '=' or '=='
+     */
+    private static final String BASE64_MATCHER_REGEX = "^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$";
```

Comment:
and now I found something that solves our problem out of the box https://docs.oracle.com/javase/8/docs/api/java/util/Base64.Decoder.html so try-catch can be used instead of regexp

Reply:
The decoder is not as strict as the RegEx.
It considers Strings with the wrong length (missing padding with `=`) as legal. Where as the RegEx also checks for the padding.
But for now it will solve our most pressing issues.

Reply:
then please lib, not regexp